plugins/deep-link/src/lib.rs

```diff
@@ -105,10 +154,164 @@ mod imp {
+                                exec = exec
+                            )
+                            .as_bytes(),
+                        )?;
+
+                        Command::new("update-desktop-database")
+                            .arg(target)
+                            .status()?;
+                    }
+
+                    Command::new("xdg-mime")
+                        .args(["default", &file_name, _protocol.as_ref()])
+                        .status()?;
+                }
+            }
+
+            Ok(())
+        }
+
+        /// Unregister the app as the default handler for the specified protocol.
+        ///
+        /// - `protocol`: The name of the protocol without `://`.
+        ///
+        /// ## Platform-specific:
+        ///
+        /// - **macOS / Linux / Android / iOS**: Unsupported.
+        pub fn unregister<S: AsRef<str>>(&self, _protocol: S) -> crate::Result<()> {
+            #[cfg(windows)]
+            {
+                let hkcu = RegKey::predef(HKEY_CURRENT_USER);
+                let base = Path::new("Software")
+                    .join("Classes")
+                    .join(_protocol.as_ref());
+
+                hkcu.delete_subkey_all(base)?;
+            }
+
+            Ok(())
+        }
+
+        /// Check whether the app is the default handler for the specified protocol.
+        ///
+        /// - `protocol`: The name of the protocol without `://`.
+        ///
+        /// ## Platform-specific:
+        ///
+        /// - **macOS / Android / iOS**: Unsupported, always returns `Ok(true)`.
+        pub fn is_registered<S: AsRef<str>>(&self, _protocol: S) -> crate::Result<bool> {
+            #[cfg(windows)]
+            {
+                let hkcu = RegKey::predef(HKEY_CURRENT_USER);
+
+                let cmd_reg = hkcu.open_subkey(format!(
+                    "Software\\Classes\\{}\\shell\\open\\command",
+                    _protocol.as_ref()
+                ))?;
+
+                let registered_cmd: String = cmd_reg.get_value("")?;
+
+                let exe = tauri::utils::platform::current_exe()?
+                    .display()
+                    .to_string()
+                    .replace("\\\\?\\", "");
+
+                Ok(registered_cmd == format!("{} \"%1\"", &exe))
+            }
+            #[cfg(target_os = "linux")]
+            {
+                let file_name = format!(
+                    "{}.desktop",
+                    tauri::utils::platform::current_exe()?
+                        .file_name()
+                        .unwrap()
+                        .to_string_lossy()
+                );
+
+                let output = Command::new("xdg-mime")
+                    .args(["default", &file_name, _protocol.as_ref()])
+                    .output()?;
+
+                Ok(String::from_utf8_lossy(&output.stdout).contains(&file_name))
+            }
+
+            #[cfg(not(any(windows, target_os = "linux")))]
+            Ok(true)
```

Comment:
My idea was that this would make cross platform handling easier like !isRegistered ? register
I see how that could cause problems too but it's similar for false i guess

Maybe a new Error::UnsupportedPlatform or would that be too harsh? At least it would be easy to spot if it returns true/false because of its actual registration status.

Reply:
> Maybe a new Error::UnsupportedPlatform or would that be too harsh? At least it would be easy to spot if it returns true/false because of its actual registration status.

maybe `Option<bool>` ?

Reply:
That only works if we're really only talking about this api, and not syncing behavior with un/register (which we should consider imo)

Reply:
if other APIs gonna return an `Error::UnsupportedPlatform` then this one should too